File: 1288-remove-covered-intervals/1288-remove-covered-intervals.py
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: (x[0], -x[1]))
        
        total_remaining = len(intervals)
        max_right_boundary = 0
        
        for current in intervals:
            current_end = current[1]
            
            if current_end <= max_right_boundary:
                total_remaining -= 1
            else:
                max_right_boundary = current_end
                
        return total_remaining

File: 1288-remove-covered-intervals/1288-remove-covered-intervals.cpp
class Solution {
public:
    int removeCoveredIntervals(vector<vector<int>>& intervals) {
        auto compareIntervals = [](const vector<int>& left, const vector<int>& right) {
            return left[0] == right[0] ? left[1] > right[1]: left[0] < right[0];
        };

        sort(intervals.begin(), intervals.end(), compareIntervals);

        int totalRemaining = intervals.size();
        int maxRightBoundary = 0;

        for (const auto& current : intervals) {
            if (current[1] <= maxRightBoundary) {
                totalRemaining--;
            } else {
                maxRightBoundary = current[1];
            }
        }

        return totalRemaining;
    }
};